docs/tutorial.rst

```diff
@@ -1,4 +1,209 @@
+  and the full time zone code.
+  The location name will be displayed to the user,
+  whereas the full code will be transmitted as the data.
+  The entries will be sorted by location name.
+
+
+The ``home.html`` template is given below.
+In the selection boxes,
+for each element in the ``zones`` list that is passed as parameter,
+the ``location`` key is used for display
+and the ``zone`` key is used for the value:
+
+.. code-block:: html
+
+   <!DOCTYPE html>
+   <html lang="en">
+   <head>
+     <meta charset="utf-8">
+     <title>Time zone convertor</title>
+   </head>
+   <body>
+     <h1>Time zone convertor</h1>
+     <form action="." method="post">
+       <select name="src">
+         {#zones}
+         <option value="{zone}">{location}</option>
+         {/zones}
+       </select>
+       <input type="datetime-local" name="dt">
+       <select name="dst">
+         {#zones}
+         <option value="{zone}">{location}</option>
+         {/zones}
+       </select>
+       <button type="submit">Show</button>
+     </form>
+   </body>
+   </html>
+
+
+Assuming you've named the Python file ``tzconvert.py``,
+when you run the command ``python tzconvert.py``,
+you can visit the address ``http://127.0.0.1:5000/``
+to see the form.
+
+
+Handling request data
+---------------------
+
+Our form submits the data to the current URL,
+therefore to the same endpoint.
+Now we want to modify the ``home()`` function,
+so that if any such data is submitted,
+the response page will include the result of the conversion:
+
+.. code-block:: python
+
+   def home(request):
+       data = {}
+       if "dt" in request.values:
+           dt = request.values.get("dt")
+           dt_naive = parser.parse(dt)
+
+           src = request.values.get("src")
+           data["src_location"] = src.split("/")[-1]
+
+           src_zone = tz.gettz(src)
+           src_dt = dt_naive.replace(tzinfo=src_zone)
+           data["src_dt"] = src_dt.ctime()
+
+           dst = request.values.get("dst")
+           data["dst_location"] = dst.split("/")[-1]
+
+           dst_zone = tz.gettz(dst)
+           dst_dt = src_dt.astimezone(dst_zone)
+           data["dst_dt"] = dst_dt.ctime()
+
+       zone_info = zoneinfo.get_zonefile_instance()
+       zone_names = zone_info.zones.keys()
+       entries = ((zone.split("/")[-1], zone) for zone in zone_names)
+       zones = [
+           {"location": location.replace("_", " "), "zone": zone}
+           for location, zone in sorted(entries)
+       ]
+       data["zones"] = zones
+       return data
+
+
+The changes are that the function now takes ``request`` as a parameter,
+and passes extra data to the template
+if the request contains a date and time information to convert.
+In the template, we add the markup for showing the result:
+
+.. code-block:: html
+
+   <body>
+     <h1>Time zone convertor</h1>
+     <p class="info">
+       When it's {src_dt} in {src_location},<br>
+       it's {dst_dt} in {dst_location}.
+     </p>
```

Comment:
If I'm understanding correctly, as written, this will always display, even if these values are empty. Probably need a conditional around this block for the first page load. Either that or defaults in the endpoint.

Reply:
Yes. I had that right in the first example I've sent but overlooked it when writing the tutorial. In my defense, it was about 3:00 AM when I wrote that part :) Anyway, it's not needed anymore now that the routes have been separated.